File: backend/components/metiers/administration_publique/tests/samples/sample_utils.test.py
# sample_utils.test.py – Tests unitaires ultra avancés pour sample_utils.py
from .sample_utils import sample_user, sample_audit_action, sample_permission_check

def test_sample_user():
    user = sample_user()
    assert user['id'].startswith('user-')
    assert user['username'] == 'sampleuser'
    assert '@' in user['email']
    assert 'admin' in user['roles']

def test_sample_audit_action():
    log = sample_audit_action()
    assert log['action'] == 'SAMPLE_ACTION'
    assert 'user' in log
    assert 'meta' in log
    assert '[FR]' in log['meta']['lang']

def test_sample_permission_check():
    assert sample_permission_check() is True

File: backend/components/metiers/administration_publique/tests/samples/sample_utils.test.js
// sample_utils.test.js – Tests unitaires ultra avancés pour sample_utils.js
const sample = require('./sample_utils');

describe('sample_utils – Génération d’utilisateur', () => {
  it('génère un utilisateur conforme', () => {
    const user = sample.sampleUser();
    expect(user).toHaveProperty('id');
    expect(user).toHaveProperty('username', 'sampleuser');
    expect(user).toHaveProperty('email');
    expect(Array.isArray(user.roles)).toBe(true);
  });
});

describe('sample_utils – Audit action', () => {
  it('génère un log d’audit conforme', () => {
    const log = sample.sampleAuditAction();
    expect(log).toHaveProperty('action', 'SAMPLE_ACTION');
    expect(log).toHaveProperty('user');
    expect(log).toHaveProperty('meta');
    expect(log.meta.lang).toMatch('[FR]');
  });
});

describe('sample_utils – Permission check', () => {
  it('vérifie la permission audit', () => {
    expect(sample.samplePermissionCheck()).toBe(true);
  });
});
